concordium-consensus/src/Concordium/Skov/Update.hs

```diff
@@ -272,24 +272,30 @@ addBlock block txvers = do
         lfs <- getLastFinalizedSlot
         -- The block must be later than the last finalized block
         if lfs >= blockSlot block then deadBlock else do
-            parentStatus <- getBlockStatus parent
+            -- Look up the block parent, if any.
+            -- This is performance sensitive since it is before we managed to validate the block,
+            -- so we make sure to look up as little data as possible to determine block validity.
+            -- In particular we do not use getBlockStatus since that loads the entire block
+            -- from the database if the block is finalized.
```

Comment:
I don't think this commentary is useful here.

Reply:
I think it does bring value since it points out that specific issue that needs care and that that is why the functionality is implemented in a way that is slightly more cumbersome than what the reader might expect.